### ISSUE TITLE
wrap 2d and wrap 3d nodes

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#658]
**Describe the project you are working on:**
spaceship game

**Describe the problem or limitation you are having in your project:**
seamless wrapping with camera's and parallax layers isn't really feasible do to the following issues..
- object colliders and graphics can't be simulated at all edges
- parallax layers aren't infinite for animated sprites and tile maps
- parallax layers jump around when the camera wraps around a certain point
- positions are vectors and can only hold so many numbers (aka true infinite space isn't possible)
- sound needs some sort of wrapping equivalent 

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
make a wrap 2d and wrap 3d node that has the following options..
- wrap boundaries 2d and 3d
- a bool to wrap its children at all possible edges (2d 8) (3d 26)
- a bool to render seamless camera edges (2d 8) (3d 26)
- a bool to translate all children relative to its wrap boundaries (including physics)
- a method to translate 2d and 3d coords to its wrapped coord system

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
basically it would just be a node that processes rendering, sound, collision, and physics of its children but in a way that doesn't break the engine.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
no this is cannot be worked around without rendering, translating and duplicating everything which is very difficult to do and requires fighting the godot engine.

**Is there a reason why this should be core and not an add-on in the asset library?:**
see above

**Comment by Shadowblitz16** [comment]
I made a more lightweight issue here if people think this isn't a good thing to add to the core.
https://github.com/godotengine/godot-proposals/issues/825

**Comment by Shadowblitz16** [comment]
I actually think a node like this should work like a chunk system except it would just be a endless grid with the same cell value repeated to the camera's edges.

basically like 3d or 4d torus of game rooms

**Comment by Shadowblitz16** [comment]
This is still relevant.

**Comment by dyrkuwu** [comment]
its still very much relevant
